java/main/src/main/java/com/saucelabs/saucebindings/VisualSession.java

```diff
@@ -0,0 +1,85 @@
+package com.saucelabs.saucebindings;
+
+import com.saucelabs.saucebindings.options.SauceOptions;
+import com.saucelabs.saucebindings.options.VisualOptions;
+import org.openqa.selenium.InvalidArgumentException;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class VisualSession extends SauceSession {
+    private VisualResults visualResults;
+    private final VisualOptions visualOptions;
+
+    public VisualSession(String testName) {
+        this(new VisualOptions(SauceOptions.chrome().setName(testName).build()));
+    }
+
+    public VisualSession(VisualOptions options) {
+        super(options.getSauceOptions());
+        this.visualOptions = options;
+    }
+
+    @Override
+    public RemoteWebDriver start() {
+        this.driver = createRemoteWebDriver(getSauceUrl(), visualOptions.toCapabilities());
+        newVisualTest(getSauceOptions().sauce().getName());
+        return driver;
+    }
+
+    @Override
+    public URL getSauceUrl() {
+        try {
+            return new URL("https://hub.screener.io/wd/hub");
+        } catch (MalformedURLException e) {
+            throw new InvalidArgumentException("Invalid URL", e);
+        }
+    }
+
+    public void newVisualTest(String testName) {
+        validateVisualStatus();
+        driver.executeScript("/*@visual.init*/", testName);
```

Comment:
shall we first verify is the session is active/driver is initialised? I assume customers might be confused by NPE if they call this method without invoking start() first

Reply:
same comment to the below methods. They only make sense while the session is active